src/core/pyproxy.c

```diff
@@ -1244,14 +1251,19 @@ _pyproxy_get_buffer(buffer_struct* target, PyObject* ptrobj)
   return 0;
 }
 
+// clang-format off
 EM_JS_REF(JsRef,
-          pyproxy_new_ex,
-          (PyObject * ptrobj, bool capture_this, bool roundtrip),
-          {
-            return Hiwire.new_value(Module.pyproxy_new(ptrobj, {
-              props : { captureThis : !!capture_this, roundtrip : !!roundtrip }
-            }));
-          });
+pyproxy_new_ex,
+(PyObject * ptrobj, bool capture_this, bool roundtrip, bool gc_register),
+{
+  return Hiwire.new_value(
+    Module.pyproxy_new(ptrobj, {
+      props: { captureThis: !!capture_this, roundtrip: !!roundtrip },
+      dontGCRegister: !gc_register,
```

Comment:
The point of the `props` is to allow proxies to share the same lifetime but have different behaviors, for instance if one is bound and the other is not. E.g., instead of:
```js
const f_unbound = pyodide.globals.get("f");
const f = f_unbound.bind(target);
f_unbound.destroy();
```
you can just do:
```js
const f = pyodide.globals.get("f").bind(target);
```
This is important because existing libraries that use `bind` don't know about `destroy`.

Anything that is not for this shouldn't go into props. I think I added a longer comment about this in a recent commit or in one of my open PRs.